tools/code_format/config.yaml

```diff
@@ -246,6 +246,7 @@ paths:
   raw_try:
     include:
     - source/common/common/thread.h
+    - source/exe/terminate_handler.cc
```

Comment:
@tyxia FYI -- thanks for being on top of thread-catching issues.

I think it does make sense to exclude terminate_handler.cc from the lint check because it will re-throw. I do expect this to run on the data plane, which is where we don't generally catch exceptions.

One thought I just had, though, @dmitri-d -- we really don't want to establish a pattern of circumventing the lint check when it's inconvenient. But I think in this case what's really happening is a re-throw after the catch, which is OK.

WDYT of instead of disabling the 'catch' check by listing the file here, we instead put a RETHROW macro in source/common/common/thread.h ? That would be cleaner IMO and avoid reinforcing a pattern of circumventing checks.

Reply:
I feel like this is a very special case: it's just trying to log some extra information before dying, due to an unknown bug (I think it's safe to say it's always a bug to have an uncaught exception in envoy). I'd be ok just leaving this as-is. Nobody would ever look at this code and think it's a pattern to copy/paste elsewhere in the code.

Reply:
I disagree -- it's not a circumvention of the check, but rather a deliberate skip. My understanding is that while the exact behaviour is unspecified, all implementations will end up calling the handler on the same thread where the exception was thrown, which means we can end up re-throwing an exception on a worker thread -- the exact thing TRY macros are guarding against.